### DISCUSSION TITLE
Could a custom schema validator be supported ?

### DISCUSSION THREAD

**Discussion by njustysq** [graphql-java/graphql-java#2249] [General]
Hi, are there plans to allow customization of schema validator ?  In some cases, I want to extend the schema validation rules. But schema validator was created while building schema. So there is no way to extend the verification rules.

```
public class GraphQLSchema {
        ...

        /**
         * Builds the schema
         *
         * @return the built schema
         */
        public GraphQLSchema build() {
            return buildImpl(false);
        }

        GraphQLSchema buildImpl(boolean afterTransform) {
            ...
            
            GraphQLSchema graphQLSchema = new GraphQLSchema(tempSchema, codeRegistry);
            schemaUtil.replaceTypeReferences(graphQLSchema);
            Collection<SchemaValidationError> errors = new SchemaValidator().validateSchema(graphQLSchema);
            if (errors.size() > 0) {
                throw new InvalidSchemaException(errors);
            }
            return graphQLSchema;
        }
    }
}
```

**Comment by bbakerman** [comment]
Out of interest what kind of rules would you put in place?

We could invent a `SchemaValidation` interface and have the above as some sort of `DefaultSchemaValidation` implementation.

This would allow someone to replace / compose a new one

But I am ask for more use cases on what type of things someone might want to put in place
